src/adc.c

```diff
@@ -19,25 +19,34 @@ static struct adc_channel * pending_measurements[ADC_QUEUE_SIZE];
 // full is a binary flag to indicate if the circular queue is full or not
 static uint8_t start, end, full;
 
+
+// Initialized the QueueMeasurement functions. Should be initialized in the header file
+// however when attempted, it didn't work, need to try again
+void QueueMeasurement(struct adc_channel * channel);
+
+// Initialized the CallCallback functions. Should be initialized in the header file
+// however when attempted, it didn't work, need to try again
+static void CallCallback(struct adc_channel * channel_struct);
+
 void ADC_Init(void) {
 	num_channels = 0;
 	start = 0;
 	end = 0;
 	full = 0;
-	hal_ADC_Init(void);
+	hal_ADC_Init();
 }
 
-void ADC_AddChannel(uint8_t channel; uint16_t period, void(*callback)(uint16_t, void *), void * ptr) {
+void ADC_AddChannel(uint8_t channel, uint16_t period, void(*callback)(uint16_t, void *), void * ptr) {
 	if(num_channels < ADC_MAX_CHANNELS) {
-		channels[num_channels].channel = channel
+		channels[num_channels].channel = channel;
 		channels[num_channels].callback = callback;
 		channels[num_channels].ptr = ptr;
 		Task_Schedule(QueueMeasurement, &channels[num_channels], period, period);
 		num_channels++;
 	}
 }
 
-static void QueueMeasurement(struct adc_channel * channel) {
+void QueueMeasurement(struct adc_channel * channel) {
```

Comment:
Why did you remove `static` here?  Also applies to the function definition above.

(This comment is to test out review comments)